src/backend/cc.d

```diff
@@ -0,0 +1,1674 @@
+                                   // a Symbol table[]
+    SFLmark         = 8,           // temporary marker
+    SFLexit         = 0x10,        // tyfunc: function does not return
+                                   // (ex: exit,abort,_assert,longjmp)
+    SFLtrue         = 0x200,       // value of Symbol != 0
+    SFLreplace      = SFLmark,     // variable gets replaced in inline expansion
+    SFLskipinit     = 0x10000,     // SCfield, SCmember: initializer is skipped
+    SFLnodebug      = 0x20000,     // don't generate debug info
+    SFLwasstatic    = 0x800000,    // was an uninitialized static
+    SFLweak         = 0x1000000,   // resolve to NULL if not found
+    SFLartifical    = 0x4000000,   // compiler generated symbol
+
+    // CPP
+    SFLnodtor       = 0x10,        // set if destructor for Symbol is already called
+    SFLdtorexp      = 0x80,        // Svalue has expression to tack onto dtor
+    SFLmutable      = 0x100000,    // SCmember or SCfield is mutable
+    SFLdyninit      = 0x200000,    // symbol has dynamic initializer
+    SFLtmp          = 0x400000,    // symbol is a generated temporary
+    SFLthunk        = 0x40000,     // symbol is temporary for thunk
+
+    // Possible values for protection bits
+    SFLprivate      = 0x60,
+    SFLprotected    = 0x40,
+    SFLpublic       = 0x20,
+    SFLnone         = 0x00,
+    SFLpmask        = 0x60,        // mask for the protection bits
+
+    SFLvtbl         = 0x2000,      // VEC_VTBL_LIST: Symbol is a vtable or vbtable
+
+    // OPTIMIZER and CODGEN
+    GTregcand       = 0x100,       // if Symbol is a register candidate
+    SFLdead         = 0x800,       // this variable is dead
+    GTunregister    = 0x2000000,   // 'unregister' a previous register assignment
+
+    // OPTIMIZER only
+    SFLunambig      = 0x400,       // only accessible by unambiguous reference,
+                                   // i.e. cannot be addressed via pointer
+                                   // (GTregcand is a subset of this)
+                                   // P.S. code generator turns off this
+                                   // flag if any reads are done from it.
+                                   // This is to eliminate stores to it
+                                   // that are never read.
+    SFLlivexit      = 0x1000,      // live on exit from function
+    SFLnotbasiciv   = 0x4000,      // not a basic induction variable
+    SFLnord         = SFLdouble,   // SCauto,SCregister,SCtmp: disallow redundant warnings
+
+    // CODGEN only
+    GTtried         = SFLmark,     // tried to place in register
+    GTbyte          = 0x8000,      // variable is sometimes accessed as
+    SFLread         = 0x40000,     // variable is actually read from
+                                   // (to eliminate dead stores)
+    SFLspill        = 0x80000,     // only in register part of the time
+}
+
+struct Symbol
+{
+//#ifdef DEBUG
+    debug ushort      id;
```

Comment:
This makes me very worried.  We shouldn't be changing the struct size and members offsets when debug is active.